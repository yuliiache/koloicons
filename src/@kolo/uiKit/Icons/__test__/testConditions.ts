export const iconWrapperProps = {
  classNames: 'test-class',
  width: '24',
  height: '56',
  viewBox: '0 0 56 56',
} as const;
