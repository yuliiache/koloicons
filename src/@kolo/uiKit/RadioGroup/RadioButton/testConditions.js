export const testRadioButtonProps = [
  {
    option: {
      id: '1',
      name: 'radioGroup',
      value: 'line',
      label: 'line',
    },

    isSelected: true,
    onChange: jest.fn(),
  },
  {
    option: {
      id: '2',
      name: 'radioGroup',
      value: 'solid',
      label: 'solid',
    },
    isSelected: false,
    onChange: jest.fn(),
  },
];
