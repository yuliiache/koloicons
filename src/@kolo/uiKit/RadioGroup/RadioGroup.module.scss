@import "@kolo/design";

.radio {
  &Group {
    display: inline-flex;
    align-items: flex-start;
    margin: 0;
    padding: 2px;
    background-color: $BrandColor;
    border-radius: 4px;
  }
}
