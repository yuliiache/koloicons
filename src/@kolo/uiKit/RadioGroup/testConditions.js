export const testRadioGroupProps = Object.freeze({
  options: [
    {
      id: '1',
      name: 'radioGroup',
      value: 'line',
      label: 'line',
    },
    {
      id: '2',
      name: 'radioGroup',
      value: 'solid',
      label: 'solid',
    },
  ],
  currentValue: 'line',
  onChange: jest.fn(),
});
