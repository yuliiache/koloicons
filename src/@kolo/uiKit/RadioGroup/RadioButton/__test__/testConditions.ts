import {Option} from '@kolo/uiKit/RadioGroup/RadioButton/RadioButton';

interface Props {
  option: Option;
  isSelected: boolean;
  onChange: jest.Mock;
}

export const testRadioButtonProps: Props[] = [
  {
    option: {
      id: '1',
      name: 'radioGroup',
      value: 'line',
      label: 'line',
    },

    isSelected: true,
    onChange: jest.fn(),
  },
  {
    option: {
      id: '2',
      name: 'radioGroup',
      value: 'solid',
      label: 'solid',
    },
    isSelected: false,
    onChange: jest.fn(),
  },
];
