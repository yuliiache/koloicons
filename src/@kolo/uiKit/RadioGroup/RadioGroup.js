import React from 'react';
import PropTypes from 'prop-types';

import RadioButton from './RadioButton/RadioButton';

import styles from './RadioGroup.module.scss';

const {radioGroup} = styles;

const RadioGroup = ({options, currentValue, onChange}) => {
  const handleOnChange = (e) => {
    onChange(e.target.value);
  };

  return (
    <div className={radioGroup}>
      {options.map((option) => (
        <RadioButton
          key={option.id}
          option={option}
          isSelected={currentValue === option.value}
          onChange={handleOnChange}
        />
      ))}
    </div>
  );
};

RadioGroup.propTypes = {
  options: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
      name: PropTypes.string,
      value: PropTypes.string,
      label: PropTypes.string,
    }).isRequired
  ),
  currentValue: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
};

export default RadioGroup;
