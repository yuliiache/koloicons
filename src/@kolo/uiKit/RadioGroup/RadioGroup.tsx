import React from 'react';

import RadioButton, {Option} from './RadioButton/RadioButton';
import styles from './RadioGroup.module.scss';

const {radioGroup} = styles;

interface Props {
  options: Option[];
  currentValue: string;
  onChange: (value: string) => void;
}
const RadioGroup: React.FC<Props> = ({options, currentValue, onChange}) => {
  const handleOnChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    onChange(e.target.value);
  };

  return (
    <div className={radioGroup}>
      {options.map((option) => (
        <RadioButton
          key={option.id}
          option={option}
          isSelected={currentValue === option.value}
          onChange={handleOnChange}
        />
      ))}
    </div>
  );
};

export default RadioGroup;
