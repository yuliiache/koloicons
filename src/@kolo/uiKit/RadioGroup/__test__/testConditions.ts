import {Option} from '@kolo/uiKit/RadioGroup/RadioButton/RadioButton';

interface Props {
  options: Option[];
  currentValue: string;
  onChange: jest.Mock;
}
export const testRadioGroupProps: Props = Object.freeze({
  options: [
    {
      id: '1',
      name: 'radioGroup',
      value: 'line',
      label: 'line',
    },
    {
      id: '2',
      name: 'radioGroup',
      value: 'solid',
      label: 'solid',
    },
  ],
  currentValue: 'line',
  onChange: jest.fn(),
});
