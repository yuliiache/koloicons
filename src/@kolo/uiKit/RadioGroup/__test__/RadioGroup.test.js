import React from 'react';
import {cleanup, render} from '@testing-library/react';

import RadioGroup from '../RadioGroup';

import {testRadioGroupProps} from './testConditions';

describe('RadioGroup', () => {
  afterEach(cleanup);

  it('should render snapshot successfully', () => {
    const {asFragment} = render(<RadioGroup {...testRadioGroupProps} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
