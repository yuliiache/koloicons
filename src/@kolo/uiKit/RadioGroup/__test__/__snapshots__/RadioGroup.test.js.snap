// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioGroup should render snapshot successfully 1`] = `
<DocumentFragment>
  <div
    class="radioGroup"
  >
    <div
      class="radioButton radioButtonActive"
      tabindex="0"
    >
      <input
        checked=""
        class="radioButtonInput"
        data-testid="radioGroupComponent"
        id="1"
        name="radioGroup"
        type="radio"
        value="line"
      />
      <label
        class="radioButtonLabel"
        for="1"
      >
        line
      </label>
    </div>
    <div
      class="radioButton"
      tabindex="0"
    >
      <input
        class="radioButtonInput"
        data-testid="radioGroupComponent"
        id="2"
        name="radioGroup"
        type="radio"
        value="solid"
      />
      <label
        class="radioButtonLabel"
        for="2"
      >
        solid
      </label>
    </div>
  </div>
</DocumentFragment>
`;
